### PR TITLE
Update fp-history.json

### PR DESCRIPTION
**Pull request by ggreif**
Mostly typos and spelling fixes.

Some changes are hiding in the long lines and the `Files changed` tab won't highlight them. I'll add a textual diff for visibility.

### CODE REVIEW

**Comment by robstewart57** [comment]
@ggreif Hi Gabor. Many thanks for this PR. What does your commit change to the text?

**Comment by ggreif** [comment]
Hi Rob, I am just recording on the fly. However I am not sure (yet) if I am working on a generated copy or on the original ;-)

~~So please don't merge yet...~~

I have transferred the changes over.

GitHub misses the changes due to long lines. If you do
```
git clone -b patch-1 git@github.com:ggreif/haflang.github.io.git
git show > pr.diff
```
and open `pr.diff` in Emacs, than you can clearly observe my edits.

Thanks for the presentation @robstewart57, very insightful!

File: docs/data/fp-history.json [hunk 1]
```diff
@@ -293,7 +293,7 @@
...
+            "text": "<p>Thomas Johnsson at Chalmers University proposed the G-Machine. It separates its abstract intermediate language, designed for graph reduction e.g. EVAL, UNWIND, PUSHFUN and UPDATE, from CPU instruction set backends. Lazy functions were compiled to G-Machine instruction sequences, which were then compiled to VAX11 CPU instruction. This work later influences an abandonment of special purpose functional hardware in the early 1990s, in favour of compiling lazy functional code to general purpose CPUs e.g. GHC's use of <a href=\"https://doi.org/10.1145/99370.99385\">STG</a>.<br><a href=\"https://doi.org/10.1145/502874.502880\">https://doi.org/10.1145/502874.502880</a></p>"
```
Comment (ggreif) [inline docs/data/fp-history.json]:
```suggestion
            "text": "<p>Thomas Johnsson at Chalmers University proposed the G-Machine. It separates its abstract intermediate language, designed for graph reduction e.g. EVAL, UNWIND, PUSHFUN and UPDATE, from CPU instruction set backends. Lazy functions were compiled to G-Machine instruction sequences, which were then compiled to VAX11 CPU instructions. This work later influences an abandonment of special purpose functional hardware in the early 1990s, in favour of compiling lazy functional code to general purpose CPUs e.g. GHC's use of <a href=\"https://doi.org/10.1145/99370.99385\">STG</a>.<br><a href=\"https://doi.org/10.1145/502874.502880\">https://doi.org/10.1145/502874.502880</a></p>"
```

**Comment by robstewart57** [comment]
@ggreif Hi Gabor, many thanks for identifying these typos, I really appreciate it.

Thanks for submitting this PR. Indeed, the files in `docs` are generated by `hakyll`. So I've created a commit reflecting the typos you spotted, and I've acknowledged your PR in the commit.

https://github.com/haflang/haflang.github.io/commit/cf023322ed4478a5d3a4c796f86f137f80d501bd

**Comment by ggreif** [comment]
My pleasure! So in the end I patched the wrong file, damn :-\

But the two files are sufficiently similar that `patch` worked for me. I hope you didn't waste too much time...

**Pull request closed without merging**